src/server/plugins/addSecurityHeaders.js

```diff
@@ -14,11 +14,39 @@
  * permissions and limitations under the License.
  */
 
-export default function addSecurityHeaders(req, res, next) {
-  res.set('X-Frame-Options', 'DENY');
-  res.set('X-Content-Type-Options', 'nosniff');
-  res.set('Strict-Transport-Security', 'max-age=15552000; includeSubDomains');
-  res.set('X-XSS-Protection', '1; mode=block');
-  res.set('Referrer-Policy', process.env.ONE_REFERRER_POLICY_OVERRIDE || 'same-origin');
-  next();
-}
+import fp from 'fastify-plugin';
+
+/**
+ * Fastify Plugin that adds security into headers
+ * @param {import('fastify').FastifyInstance} fastify Fastify instance
+ * @param {import('fastify').FastifyPluginOptions} _opts plugin options
+ * @param {import('fastify').FastifyPluginCallback} done plugin callback
+ */
+const addSecurityHeaders = (fastify, opts = {}, done) => {
+  const ignoreRoutes = opts.ignoreRoutes || [];
+
+  fastify.addHook('onRequest', async (request, reply) => {
```

Comment:
Do you need async here?

Reply:
it is, fastify detects it and removes the need to call the `done` fn